.github/workflows/action-image-scan-integration-dashing.yaml

```diff
@@ -0,0 +1,22 @@
+name: "Check docker image vulnerabilities"
+
+on: 
+  pull_request:
+  workflow_dispatch:
+
+env:
+  ROS_VERSION: 2
+  UBUNTU_DISTRO: bionic
+  ROS_DISTRO: dashing
+
+jobs:
+  build-secure-and-push:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - run: docker build --build-arg ROS_VERSION=$ROS_VERSION --build-arg UBUNTU_DISTRO=$UBUNTU_DISTRO --build-arg ROS_DISTRO=$ROS_DISTRO --target devel-build -t ros$ROS_VERSION-$ROS_DISTRO:devel .
+      
+    - uses: Azure/container-scan@v0
+      with:
+        image-name: ros${{ env.ROS_VERSION }}-${{ env.ROS_DISTRO }}:devel
```

Comment:
Why did you create a separate job for this? Isn't it the same as the first steps of the deployment test?

Reply:
Additionally, I think this job can share the same file as the deployment job

Reply:
Deployment, needs deploy the images to docker hub, I put to do the same as push, just as validation, and I was trying to see how the badges would work

Reply:
> Additionally, I think this job can share the same file as the deployment job

I will check your matrix comment

Reply:
Concerning the use of matrix we have a problem that it does not generate separated badges, and I would like to have a badge for each ros-distro that we support, at least on deploy. So I added the matrix setup on the integration code, however for deploy I am using two actions, so we have ros-distros separately, check 8274b3cdc453a9a8844a8b335d39fa41efd8b20b

Reply:
Also, see [here](https://github.community/t/separate-workflow-badges-when-using-matrix-testing-possible/16708/5)

Reply:
Nice. I also like to have badges for each distribution.